### ISSUE TITLE
[Refactor] Refactoring `/message` endpoint in order to execute the agent

### ISSUE THREAD

**Issue by valentimarco** [cheshire-cat-ai/core#921]
At the moment, `/message` is mostly bugged when trying to interact with the agent.

The last dev meeting, we discuss how can be:
1. No streaming (If you need it, use WS.). The reason is simple: This http method will be design for agent interaction that are not conversational but rather utility (Translations, Interaction with CRMs etc...)
2. JSON that contains at least the key "text" (like we do in WS). This way if you have some custom stuff that reads the json, you are free to do it even in http.
3. Some python types to help OpenApi gen (PLS)

**Comment by MihirKohli** [comment]
Can you please brief more about the issue.

**Comment by mastrogiovanni** [comment]
What to do with websocket notifications? 
I fixed the /message endpoint with the following (mutuated from ws interface):
```python
@router.post("/message", response_model=CatMessage)
async def message_with_cat(
    payload: Dict = Body({"text": "hello!"}),
    stray=Depends(HTTPAuth(AuthResource.CONVERSATION, AuthPermission.WRITE)),
) -> Dict:
    """Get a response from the Cat"""
    payload["user_id"] = stray.user_id
    return await stray.__call__(payload)
```

The problem is that stray cat try to send notification to websocket producing a lot of warnings.
Is that ok if I'll remove the warnings and I'll leave the current behaviour that simply ignore the notifications if ws is None?

**Comment by valentimarco** [comment]
Fixed with #932
